client/src/components/Login.js

```diff
@@ -1,106 +1,154 @@
-import React, { Component } from 'react';
-import axios from 'axios';
-import {Link} from 'react-router-dom';
-import Grid from '@material-ui/core/Grid';
-import Typography from '@material-ui/core/Typography';
-import TextField from '@material-ui/core/TextField';
+import React, { Component } from "react";
+import axios from "axios";
+import { Link, Redirect } from "react-router-dom";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import TextField from "@material-ui/core/TextField";
+import Snackbar from "@material-ui/core/Snackbar";
+import Icon from "@material-ui/core/Icon";
+import IconButton from "@material-ui/core/IconButton";
+import { withStyles } from "@material-ui/styles";
 
-import FormStyles from '../styles/FormStyles';
-import StyledButton from '../styles/StyledButton';
+import styles from "../styles/FormStyles";
+import StyledButton from "../styles/StyledButton";
 
-const ButtonLink = React.forwardRef(
-  (props, ref) => <Link innerRef={ref} {...props} />
-);
+const ButtonLink = React.forwardRef((props, ref) => (
+  <Link innerRef={ref} {...props} />
+));
 
 class Login extends Component {
-  state = { 
-    email: '',
-    password: '',
-    msg: null
-  }
+  state = {
+    email: "",
+    password: "",
+    errors: {},
+    redirect: false
+  };
 
   onChange = e => {
     this.setState({
       [e.target.name]: e.target.value
     });
-  }
+  };
 
   onSubmit = e => {
     e.preventDefault();
 
     const { email, password } = this.state;
 
-    axios.post('/api/users/login', {email, password})
+    axios
+      .post("/api/users/login", { email, password })
       .then(res => {
-        //TODO: redirect user to homepage
-        const token = res.data.token;
-        sessionStorage.setItem('token', token);
+        sessionStorage.setItem("token", res.data.token);
```

Comment:
I think it's better to store the token in the localStorage, not in the sessionStorage. Because we use token to validity the user, we do not need session any more. And the server side does not generate or store the session.